lib/algos/list/delimiterCurrent.ts

```diff
@@ -43,9 +49,57 @@ class DelimiterCurrent extends Delimiter {
                 ne: this.excludedDataStoreName,
             }
         }
+
+        this.start = Date.now();
+
         return params;
     }
 
+    _parse(s) {
+        let p;
+        try {
+            p = JSON.parse(s);
+        } catch (e: any) {
+            this.logger.warn(
+                'Could not parse Object Metadata while listing',
+                { err: e.toString() });
+        }
+        return p;
+    }
+
+    filter(obj) {
+        if (this.start && Date.now() - this.start > DELIMITER_TIMEOUT_MS) {
+            this.IsTruncated = true;
+            this.logger.info('listing stopped after expected internal timeout',
+                {
+                    timeoutMs: DELIMITER_TIMEOUT_MS,
+                    evaluatedKeys: this.evaluatedKeys,
+                });
+            return FILTER_END;
+        }
+        ++this.evaluatedKeys;
+
+        const parsedValue = this._parse(obj.value);
+        // if parsing fails, skip the key.
+        if (parsedValue) {
+            const lastModified = parsedValue['last-modified'];
+            const dataStoreName = parsedValue.dataStoreName;
+            // We then check if the current version is older than the "beforeDate" and
+            // "excludedDataStoreName" is not specified or if specified and the data store name is different.
+            if ((!this.beforeDate || (lastModified && lastModified < this.beforeDate)) && 
+            (!this.excludedDataStoreName || dataStoreName !== this.excludedDataStoreName)) {
+                return super.filter(obj);
```

Comment:
cf my comment for the review, I think this would be called for all keys (master and versions), hence the Delimiter class would output all versions of an object in the result.

Reply:
With v1 bucket format only masters are returned. I already wrote tests for it in Cloudserver development/8.x branches.

To support v0, we are planing on leveraging DelimiterMaster.